pkg/apis/eventing/v1alpha1/parameter_argument_defaults.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// SetDefaults defaults the ParameterSpec spec.
```

Comment:
I don't understand why we only have these defaults here.

Separately, I'm not sure that key-value (vs structured) arguments are the way to go with provisioners. It made more sense when we were converting these directly into key-value command-line arguments.

Reply:
the defaults method is to blindly call down to the structs from the Provisioner spec and let the sub-struct decide if it wants a default. 

custom extensions would not work with our strict json processing, how would you define this for different providers?

Reply:
You could use [`runtime.RawExtension`](https://godoc.org/k8s.io/apimachinery/pkg/runtime#RawExtension) to hold the message.

Yes, this would mean that we couldn't validate the different sets of arguments to a provisioner to make sure that they are in the right shape. It's not clear that we can usefully do that with a key-value map, either (what if one of the values is an encoded object?).